### ISSUE TITLE
[Feature Request] Glade/Qt Creator-like UI editor

### ISSUE THREAD

**Issue by leapofazzam123** [Immediate-Mode-UI/Nuklear#344]
Hi, I'm looking for lightweight UI library and found Nuklear. I found this UI library is very lightweight and want to use it in my future projects. But, since I'm a very newbie developer, I don't want to code all of the UI in a C file. I want a Glade/Qt Creator-like UI editor for Nuklear. I also want to have a custom file (like .nui) for Nuklear UI so I can just load it in my code.

**Comment by RobLoach** [comment]
Would be pretty cool, but considering it's immediate-mode, I imagine the state management being pretty tricky to manage. I'd recommend reading through the API a bit, and seeing what you could come up with for it!

**Comment by dumblob** [comment]
@RobLoach yep, it wouldn't be a piece of cake - but probably not much more difficult than for retained UIs if you think of it (of course, you must not expect it to generate similar code to what one would write by hand).

I think that such an editor is anyway just an advanced "motivational element" which will nearly in 100% cases be abandoned after the first few hours of generating the "start off template". Same as providing many backends to choose from and same as providing examples (it's orders of magnitude easier to remove stuff than to come up with "what to add" on your own).
